iiblib/iib_build_details_model.py

```diff
@@ -1,135 +1,336 @@
 class IIBBuildDetailsModel(object):
-    """Model class handling data about index build task"""
-
-    def __init__(
-        self,
-        _id,
-        state,
-        reason,
-        state_history,
-        from_index,
-        from_index_resolved,
-        bundles,
-        removed_operators,
-        organization,
-        binary_image,
-        binary_image_resolved,
-        index_image,
-        request_type,
-        arches,
-        bundle_mapping,
-        omps_operator_version,
-    ):
+    """
+    Model class handling data about index build task
+
+    Args:
+        id (int)
+            Id of build
+        arches (list)
+            List of architectures supported in new index image
+        state (str)
+            State of build
+        state_reason (str)
+            Reason for state change
+        request_type (str)
+            Type of iib build task
+        batch (int)
+            Number of batches included in the request
+        updated (str)
+            Time when was the request updated
+        user (str)
+            User kerberos (email)
+        state_history (list) - optional
+            List of directories where is state, state_reason and updated
+        batch_annotations (dict) - optional
+            Annotation of the batch
+        logs (dict) - optional
+            Dictionary contains url of the log and expiration date
+    """
+
+    __slots__ = [
+        "id",
+        "arches",
+        "state",
+        "state_reason",
+        "request_type",
+        "batch",
+        "updated",
+        "user",
+        "state_history",
+        "batch_annotations",
+        "logs",
+        "_data",
+    ]
+
+    _general_attrs = [
+        "id",
+        "arches",
+        "state",
+        "state_reason",
+        "request_type",
+        "batch",
+        "updated",
+        "user",
+    ]
+
+    _optional_attrs = {
+        "state_history": lambda: list(),
+        "batch_annotations": lambda: dict(),
+        "logs": lambda: dict(),
+    }
+
+    _operation_attrs = []
+
+    _accepted_request_type = ""
```

Comment:
Instead of this, why not have a list of accepted_request_types here and use that?

Reply:
The IIBBuildDetailsModel will accept many request types but the variable is important in child models to distinguish "request_type" and data variables from the response. The request_type has to match with a structure of data.